### PR TITLE
Qa to Master: dev mode

### PR DESCRIPTION
**Pull request by evelinec**
Converted guide to run with dev mode. The follow changes were made: 

- use the ideal(simplified) pom.xml

- update the Readme content to use dev mode

  - keep the flow same (let discuss if the flow not work)

- convert *Test class to *IT
- convert to use JUnit 5
  - use JUnit5 test order if need

- make sure to correctly configure the ports in test classes, server.xml, and pom.xml

- update travis test
  - mvn install cannot be used to verify anymore

### CODE REVIEW

**Comment by evelinec** [comment]
@Charlotte-Holt This PR is ready for you to review as well. Thanks.

File: README.adoc [hunk 1]
```diff
@@ -107,8 +95,7 @@ include::finish/src/main/resources/META-INF/microprofile-config.properties[]
 // =================================================================================================
 == Injecting static configuration
 
-The MicroProfile Config API was added as a dependency to your [hotspot file=0]`pom.xml` file. Look for the dependency with the [hotspot=config_aID file=0]`mpConfig` artifact ID.
-This feature allows you to use the MicroProfile Config API to externalize configurations for your microservices.
+The MicroProfile Config API is included in the MicroProfile dependency specified in your [hotspot file=0]`pom.xml` file. Look for the dependency with the [hotspot=microprofile file=0]`microprofile` artifact ID. This dependency provides a library that allows you to use the MicroProfile Config API to externalize configurations for your microservices.
```
Comment (Charlotte-Holt) [inline README.adoc]: Change
"The MicroProfile Config API is included in the MicroProfile dependency specified in your [hotspot file=0]`pom.xml` file."
To
"The MicroProfile Config API is included in the MicroProfile dependency that is specified in your [hotspot file=0]`pom.xml` file."

File: README.adoc [hunk 2]
```diff
@@ -384,51 +367,63 @@ include::{common-includes}/mvncompile.adoc[]
...
 ----
 
-The [hotspot=testInitialServiceStatus file=0]`testInitialServiceStatus()` test case reads the value of the `io_openliberty_guides_inventory_inMaintenance` configuration property in the file `META-INF/microprofile-config.properties` and checks the HTTP response of the inventory service.
+microprofile-config.properties
+[source, properties, linenums, role='code_column']
+----
+include::finish/src/main/resources/META-INF/microprofile-config.properties[]
+----
+
+The [hotspot=testInitialServiceStatus file=0]`testInitialServiceStatus()` test case reads the value of the [hotspot=inventory-in-maintenance file=1]`io_openliberty_guides_inventory_inMaintenance` configuration property in the file [hotspot file=1]`META-INF/microprofile-config.properties` and checks the HTTP response of the inventory service.
```
Comment (Charlotte-Holt) [inline README.adoc]: Change
"The [hotspot=testInitialServiceStatus file=0]`testInitialServiceStatus()` test case reads the value of the [hotspot=inventory-in-maintenance file=1]`io_openliberty_guides_inventory_inMaintenance` configuration property in the file [hotspot file=1]`META-INF/microprofile-config.properties` and checks the HTTP response of the inventory service."
To
"The [hotspot=testInitialServiceStatus file=0]`testInitialServiceStatus()` test case reads the value of the [hotspot=inventory-in-maintenance file=1]`io_openliberty_guides_inventory_inMaintenance` configuration property in the [hotspot file=1]`META-INF/microprofile-config.properties` file and checks the HTTP response of the `inventory` service."

File: README.adoc [hunk 3]
```diff
@@ -384,51 +367,63 @@ include::{common-includes}/mvncompile.adoc[]
...
 In the end, the inventory service returns the following message: `ERROR: Service is currently in maintenance`.
 
 The [hotspot=testChangeEmail file=0]`testChangeEmail()` test case first puts the `inventory` service in maintenance, then it changes the email address in the configuration file. In the end, the `inventory` service should display the error message with the latest email address.
 
 In addition, a few endpoint tests have been provided for you to test the basic functionality of the `inventory` and `system` services. If a test failure occurs, then you must have introduced a bug into the code.
-Reminder that you must register the custom configuration source and custom converter in the `src/main/resources/META-INF/services/` directory, if you miss to complete these steps, the tests will fail.
+Reminder that you must register the custom configuration source and custom converter in the `src/main/resources/META-INF/services/` directory, if you miss to complete these steps, the tests will fail. These tests
+will run automatically as a part of the integration test suite.
```
Comment (Charlotte-Holt) [inline README.adoc]: Change
"Reminder that you must register the custom configuration source and custom converter in the `src/main/resources/META-INF/services/` directory, if you miss to complete these steps, the tests will fail. These tests
will run automatically as a part of the integration test suite."
To
"Remember that you must register the custom configuration source and custom converter in the `src/main/resources/META-INF/services/` directory. If you don't complete these steps, the tests will fail. These tests run automatically as a part of the integration test suite."

File: README.adoc [hunk 4]
```diff
@@ -384,51 +367,63 @@ include::{common-includes}/mvncompile.adoc[]
...
+
 
 [role=command]
-include::{common-includes}/mvnverify.adoc[]
+include::{common-includes}/devmode-test.adoc[]
+
+You will see the following output:
```
Comment (Charlotte-Holt) [inline README.adoc]: Change
"You will see the following output:"
To
"You see the following output:"

File: README.adoc [hunk 5]
```diff
@@ -384,51 +367,63 @@ include::{common-includes}/mvncompile.adoc[]
...
+[INFO] Running it.io.openliberty.guides.inventory.InventoryEndpointIT
+[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.077 s - in it.io.openliberty.guides.inventory.InventoryEndpointIT
+[INFO]
+[INFO] Results:
+[INFO]
+[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0
+----
+
+To see whether the tests detect a failure, remove the configuration resetting line in the [hotspot=setup file=0]`setup()` method of the [hotspot file=0]`ConfigurationIT.java` file.
 Then manually change some configuration values in the `resources/CustomConfigSource.json` file.
```
Comment (Charlotte-Holt) [inline README.adoc]: Change
"Then manually change some configuration values in the `resources/CustomConfigSource.json` file."
To
"Then, manually change some configuration values in the `resources/CustomConfigSource.json` file."

File: README.adoc (hunk 1 above)
Comment (evelinec) [reply README.adoc]: Updated

File: README.adoc (hunk 2 above)
Comment (evelinec) [reply README.adoc]: Updated

File: README.adoc (hunk 3 above)
Comment (evelinec) [reply README.adoc]: Updated

File: README.adoc (hunk 4 above)
Comment (evelinec) [reply README.adoc]: Updated

File: README.adoc (hunk 5 above)
Comment (evelinec) [reply README.adoc]: Updated

**Comment by evelinec** [comment]
Feedback addressed in #116. @Charlotte-Holt can you take a look? Thank you.

**Comment by evelinec** [comment]
Verified ID changes and re-tested guide with the latest changes on qa site.

**Pull request merged**